bigquery_etl/metadata/validate_metadata.py

```diff
@@ -27,6 +32,72 @@ def validate_public_data(metadata, path):
     return is_valid
 
 
+def validate_change_control(
+    file_path,
+    metadata,
+    codeowners_file,
+    project_id="moz-fx-data-shared-prod",
+    sql_dir="sql",
+):
+    """Verify that a query is correctly setup for change control."""
+    path_to_add = file_path.partition(f"{project_id}/")[2]
+    path_in_codeowners = os.path.join(sql_dir, project_id, path_to_add)
+    has_change_control = CHANGE_CONTROL_LABEL in metadata.labels
+
+    if has_change_control:
+        # This label requires to have at least one owner in the metadata file.
+        # And for any of the owners, at least one entry in the CODEOWNERS file.
+        # The owners can be emails or GitHub identities e.g. mozilla/team_name.
+        if len(metadata.owners) == 0:
+            click.echo(
+                click.style(
+                    f"The metadata for {file_path} has the label"
+                    f" change_controlled but it's missing owners."
+                )
+            )
+            return False
+
+        with open(codeowners_file, "r") as owners_file:
+            content = owners_file.readlines()
```

Comment:
I suggest moving the `startswith` check here to avoid duplicating it below.

Reply:
Yes, also to not validate lines unnecessarily.